benchmarking/run.py

```diff
@@ -0,0 +1,315 @@
+from runner.matrix import MatrixConfig, MatrixEntry
+from runner.path_resolver import PathResolver
+from runner.process import run_command_with_timeout
+from runner.ray_cluster import (
+    setup_ray_cluster_and_env,
+    teardown_ray_cluster_and_env,
+)
+from runner.utils import find_result, get_obj_for_json, resolve_env_vars
+
+
+def ensure_dir(dir_path: Path) -> None:
+    """Ensure dir_path and parents exists, creating them if necessary."""
+    dir_path.mkdir(parents=True, exist_ok=True)
+
+
+def get_entry_script_persisted_data(benchmark_results_path: Path) -> dict[str, Any]:
+    """Read the files that are expected to be generated by the individual benchmark scripts."""
+    params_json = benchmark_results_path / "params.json"
+    if not params_json.exists():
+        logger.warning(f"Params JSON file not found at {params_json}")
+        script_params = {}
+    else:
+        with open(params_json) as f:
+            script_params = json.load(f)
+
+    metrics_json = benchmark_results_path / "metrics.json"
+    if not metrics_json.exists():
+        logger.warning(f"Metrics JSON file not found at {metrics_json}")
+        script_metrics = {}
+    else:
+        with open(metrics_json) as f:
+            script_metrics = json.load(f)
+
+    tasks_pkl = benchmark_results_path / "tasks.pkl"
+    if not tasks_pkl.exists():
+        logger.warning(f"Tasks pickle file not found at {tasks_pkl}")
+        script_tasks = []
+    else:
+        with open(tasks_pkl, "rb") as f:
+            script_tasks = pickle.load(f)  # noqa: S301
+        if isinstance(script_tasks, list):
+            script_metrics.update(TaskPerfUtils.aggregate_task_metrics(script_tasks, prefix="task"))
+        elif isinstance(script_tasks, dict):
+            for pipeline_name, pipeline_tasks in script_tasks.items():
+                script_metrics.update(
+                    TaskPerfUtils.aggregate_task_metrics(pipeline_tasks, prefix=pipeline_name.lower())
+                )
+
+    return {"params": script_params, "metrics": script_metrics}
+
+
+def check_requirements_update_results(result_data: dict[str, Any], requirements: dict[str, Any]) -> bool:
+    """
+    Check if the benchmark meets the requirements. Creates a new "requirements" key in the result_data
+    dictionary with the results of the requirements checks.
+    Returns True if the benchmark meets the requirements, False otherwise.
+    """
+    meets_requirements = True
+    requirements_data = {}
+
+    for metric_name, requirement_dict in requirements.items():
+        reason_not_met = None
+        actual_value = find_result(result_data, metric_name)
+        if actual_value is None:
+            reason_not_met = f"{metric_name} not found in metrics"
+        elif "min_value" in requirement_dict:
+            min_value = requirement_dict["min_value"]
+            if actual_value < min_value:
+                reason_not_met = f"{metric_name} < {min_value}"
+        elif "max_value" in requirement_dict:
```

Comment:
i think if both `min_value` and `max_value` are provided then the code might not work

Reply:
Thanks, I think I fixed it in the latest commit.